src/obj-ignore.c

```diff
@@ -273,9 +274,11 @@ int apply_autoinscription(struct player *p, struct object *obj)
 	object_desc(o_name, sizeof(o_name), obj, ODESC_PREFIX | ODESC_FULL, p);
 
 	if (note[0] != 0)
-		obj->note = quark_add(note);
-	else
-		obj->note = 0;
+		obj->note = string_make(note);
+	else {
+		string_free(obj->note);
```

Comment:
Move `string_free(obj->note);` to immediately before the `if (note[0] != 0)` test to avoid a memory leak if the auto inscription will replace an existing inscription.